odo/backends/pandas.py

```diff
@@ -57,9 +57,18 @@ def coerce_datetimes(df):
     name            object
     dtype: object
     """
-    df2 = df.select_dtypes(include=['object']).apply(
-        partial(pd.to_datetime, errors='ignore')
-    )
+
+    def patched_to_datetime(s):
+        # NOTE: In pandas < 0.17, pd.to_datetime(' ') == datetime(...), which
+        # is not what we want.  This patched version detects empty or
+        # whitespace-only strings, and maps them to a dummy object.  This
+        # prevents datetime coercion on whitespace or empty strings.
+        o = object()
+        s_norm = s.apply(lambda x: o if (hasattr(x, 'strip') and not x.strip()) else x)
```

Comment:
also don't apply pd.to_datetime it works the other way
eg pd.to_datetime(s) is vectorized